### PR TITLE
Float32 min/max/rounding intrinsics

### PR DESCRIPTION
**Pull request by TheNumbat**
Adds intrinsics for `MINSS`,`MAXSS`,`ROUNDSS`, reaching parity with float64.

Note that while `Ocaml_intrinsics` also provides `round_nearest`, the float32 library does not. 
This is because we can't easily provide a cross-platform C implementation of round-half-to-even.
Instead, users should use `round_half_to_even` (respective of the current rounding mode).

Tests have been added to `float32_lib.ml` and `float32_u_lib.ml`.

p.s. moves `Sqrt_scalar_f32` from `SSE2` to `SSE`, where it belongs.

### CODE REVIEW

File: tests/small_numbers/float32_lib.ml [hunk 1]
```diff
@@ -230,13 +235,24 @@ let () =
...
     assert((F32.equal f1 f2) = (CF32.equal f1 f2));
   )
 ;;
 
+let () =
+  CF32.check_float32s (fun f _ ->
+    bit_eq (F32.round_up f) (CF32.ceil f);
+    bit_eq (F32.round_down f) (CF32.floor f);
+    bit_eq (F32.round_half_to_even f) (CF32.round_current f);
+    assert ((F32.iround_half_to_even f) = (CF32.iround_current f));
```
Comment (xclerc) [inline tests/small_numbers/float32_lib.ml:252]: I think there should be a comment
explaining why this line is different.

Comment (TheNumbat) [reply tests/small_numbers/float32_lib.ml:253]: Added

File: ocaml/otherlibs/stdlib_beta/float32.ml [hunk 2]
```diff
@@ -267,6 +277,19 @@ let[@inline] min_max_num (x : t) (y : t) =
...
+  [@@noalloc] [@@unboxed] [@@builtin]
+
+external round_intrinsic : (int[@untagged]) -> (t[@unboxed]) -> (t[@unboxed])
+  = "caml_sse41_float32_round_bytecode" "caml_sse41_float32_round"
+  [@@noalloc] [@@builtin]
+
+let[@inline] round_half_to_even x = round_intrinsic 0xC x
+let[@inline] round_down x = round_intrinsic 0x9 x
+let[@inline] round_up x = round_intrinsic 0xA x
+let[@inline] round_towards_zero x = round_intrinsic 0xB x
```
[the remark this answers is not in the document]
Comment (TheNumbat) [reply ocaml/otherlibs/stdlib_beta/float32.ml]: Added

File: ocaml/runtime/float32.c [hunk 3]
```diff
@@ -309,6 +309,51 @@ CAMLprim value caml_ldexp_float32_bytecode(value f, value i)
...
+CAMLprim value caml_sse_float32_min_bytecode(value x, value y) {
+  return caml_copy_float32(caml_sse_float32_min(Float32_val(x), Float32_val(y)));
+}
+
+float caml_sse_float32_max(float x, float y) {
+  return x > y ? x : y;
+}
+
+CAMLprim value caml_sse_float32_max_bytecode(value x, value y) {
+  return caml_copy_float32(caml_sse_float32_max(Float32_val(x), Float32_val(y)));
```
[the remark this answers is not in the document]
Comment (TheNumbat) [reply ocaml/runtime/float32.c]: Fixed

**Review by xclerc — APPROVED** [review]

**Pull request merged**